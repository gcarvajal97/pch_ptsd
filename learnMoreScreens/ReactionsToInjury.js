import i18n from 'i18n-js';
import React, { Component } from "react";
import { View, StyleSheet, Text, ScrollView } from "react-native";
import { Video } from 'expo-av';
import translate, { getVideo } from '../components/translateService';
import colors from '../colors';

const videos = {
    en: require("../assets/videos/vidReactToInjury.mp4"),
    es: require("../assets/videos/vidReactToInjuryEs.mp4"),
  };

class ReactionsToInjury extends Component {

    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
        headerTitle: translate("reactionsToInjury.header")
    }
    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.title}>
                    {translate('reactionsToInjury.title')}
                </Text>
                <Video
                    // Import video corresponding to this screen from assets
                    source={getVideo(videos)}
                    volume={1.0}
                    resizeMode={Video.RESIZE_MODE_COVER}
                    shouldPlay
                    useNativeControls
                    style={{ 
                        height: 204, 
                        width: '100%', 
                        maxHeight: 254, 
                        borderColor: colors.secondary.pms_3005, 
                        borderWidth: 14, 
                        borderTopWidth: 0, 
                        borderBottomWidth: 0, 
                        marginBottom: -1
                    }}
                    accessible
                    accessibilityLabel={translate('reactionsToInjury.content.videoCard.accessibility')}
                    accessibilityHint={translate('reactionsToInjury.content.videoCard.accessibility')}
                />
                <Text style={styles.subTitle}>
                    <Text style={{ fontWeight: 'bold' }}>{translate('reactionsToInjury.content.videoCard.title')}</Text> 
                    {translate('reactionsToInjury.content.videoCard.subtitle')}
                </Text>
                <ScrollView>
                    <Text style={styles.paragraph}>{translate('reactionsToInjury.content.paragraphOne')}</Text>
                    <Text style={styles.paragraph}>{translate('reactionsToInjury.content.paragraphTwo')}</Text>
                    <Text style={{fontWeight: 'bold', fontSize:20, alignSelf:'stretch', textAlign:'left',marginLeft:5, marginTop:2, marginBottom:8, color:'#2089DC'}}>
                    {translate('reactionsToInjury.content.bulletListOne.title')}
                    </Text>
                    <Text style={styles.bullet}>{translate('reactionsToInjury.content.bulletListOne.bulletOne')}</Text>
                    <Text style={styles.bullet}>{translate('reactionsToInjury.content.bulletListOne.bulletTwo')}</Text>
                    <Text style={styles.bullet}>{translate('reactionsToInjury.content.bulletListOne.bulletThree')}</Text>
                    <Text style={styles.bullet}>{translate('reactionsToInjury.content.bulletListOne.bulletFour')}</Text>
                    <Text style={styles.bulletLast}>{translate('reactionsToInjury.content.bulletListOne.bulletFive')}</Text>
                    <Text style={{fontWeight: 'bold', fontSize:20, alignSelf:'stretch', textAlign:'left',marginLeft:5, marginTop:2, marginBottom:8, color:'#2089DC'}}>
                    {translate('reactionsToInjury.content.bulletListTwo.title')}
                    </Text>
                    <Text style={styles.bullet}>{translate('reactionsToInjury.content.bulletListTwo.bulletOne')}</Text>
                    <Text style={styles.bulletLast}>{translate('reactionsToInjury.content.bulletListTwo.bulletTwo')}</Text>
                    <Text style={styles.paragraph}>{translate('reactionsToInjury.content.paragraphThree')}
                    <Text style={{fontWeight:'bold'}}> {translate('reactionsToInjury.content.paragraphThreeBold')}</Text>
                    </Text>        
                </ScrollView>
            </View>
        );
    }
}
export default ReactionsToInjury;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        margin: 10,
        backgroundColor: colors.primary.white,
        shadowColor: colors.primary.black,
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.23,
        shadowRadius: 2.62,

        elevation: 4,
    },
    bullet: {
        marginVertical:1, 
        marginHorizontal:18, 
        textAlign:'left', 
        alignSelf:'stretch', 
        fontWeight:'bold',
        fontSize:14
    },
    bulletLast: {
        marginTop:1,
        marginBottom:6, 
        marginHorizontal:18, 
        textAlign:'left', 
        alignSelf:'stretch', 
        fontWeight:'bold',
        fontSize:14
    },
    paragraph: {
        alignSelf:"center", 
        paddingVertical:1, 
        paddingHorizontal:5,
        fontSize: 14
    },
    title: {
        backgroundColor: colors.accent, 
        color: colors.primary.white, 
        fontWeight: 'bold', 
        fontSize: 22, 
        textAlign: 'center', 
        alignSelf: 'stretch'
    },
    subTitle: {
        backgroundColor: colors.accent, 
        color: colors.primary.white, 
        alignSelf: 'stretch', 
        paddingLeft: 15, 
        paddingRight: 15, 
        fontSize: 15, 
        borderBottomWidth: 2, 
        borderBottomColor: colors.accent,
    }
});