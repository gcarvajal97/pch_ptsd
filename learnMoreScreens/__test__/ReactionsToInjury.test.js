import React from 'react';
import ReactionsToInjury from '../ReactionsToInjury';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<ReactionsToInjury />).toJSON();
    expect(tree).toMatchSnapshot();
})