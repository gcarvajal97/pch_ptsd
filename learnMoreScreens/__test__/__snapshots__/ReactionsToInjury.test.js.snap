// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "elevation": 4,
      "flex": 1,
      "margin": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#2089DC",
        "color": "white",
        "fontSize": 22,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Reactions to Injury
  </Text>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#2089DC",
        "borderTopWidth": 0,
        "borderWidth": 14,
        "height": 204,
        "marginBottom": -1,
        "maxHeight": 254,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      accessibilityHint="Video show what are stress reactions to injury"
      accessibilityLabel="Video show what are stress reactions to injury"
      accessible={true}
      proxiedProperties={
        Object {
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": Object {
            "headers": undefined,
            "overridingExtension": null,
            "uri": "uri",
          },
          "status": Object {
            "rate": 1,
            "shouldPlay": true,
            "volume": 1,
          },
          "useNativeControls": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#2089DC",
        "borderBottomColor": "#2089DC",
        "borderBottomWidth": 2,
        "color": "white",
        "fontSize": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      These reactions are expected.
    </Text>
     
    With a little time and extra support, most children feel better.
  </Text>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        One of the most important things you can do to help your child is to recognize your childâ€™s reactions to injury. Many reactions are normal and common; some are more worrisome.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        The first step in helping your child recover is to learn what reactions to look for and what you can do to help your child respond in a healthy way.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginLeft": 5,
            "marginTop": 2,
            "textAlign": "left",
          }
        }
      >
        In the first few days after an injury, these are the common child trauma symptoms:
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Nearly all children feel upset, jumpy, or worried at times.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Parents and other family members can have similar reactions.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Nearly everyone feels symptoms of stress; Headache, tense muscles, knots in your stomach, sweaty palms, and the feeling that life is a bit out of control.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Parents and children may argue more.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginBottom": 6,
            "marginHorizontal": 18,
            "marginTop": 1,
            "textAlign": "left",
          }
        }
      >
        - Children may be more clingy or have trouble sleeping.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginLeft": 5,
            "marginTop": 2,
            "textAlign": "left",
          }
        }
      >
        You may even notice some positive reactions to their injury:
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 18,
            "marginVertical": 1,
            "textAlign": "left",
          }
        }
      >
        - Feeling thankful and relieved that nothing worse happened
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginBottom": 6,
            "marginHorizontal": 18,
            "marginTop": 1,
            "textAlign": "left",
          }
        }
      >
        - A resolve to do things more safely in the future
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        When an accident or injury causes overwhelming feelings of fear, helplessness, and horror, it can lead to more than just everyday stress reactions. It can lead to
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
           
          traumatic stress.
        </Text>
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
